src/ophyd_async/epics/_backend/_aioca.py

```diff
@@ -147,11 +153,12 @@ def make_converter(
     else:
         value = list(values.values())[0]
         # Done the dbr check, so enough to check one of the values
-        if datatype and not isinstance(value, datatype):
-            raise TypeError(
-                f"{pv} has type {type(value).__name__.replace('ca_', '')} "
-                + f"not {datatype.__name__}"
-            )
+        if datatype:
+            if not isinstance(value, datatype):
+                raise TypeError(
+                    f"{pv} has type {type(value).__name__.replace('ca_', '')} "
+                    + f"not {datatype.__name__}"
+                )
```

Comment:
Nit: why did this change?

Reply:
It's a left over from the first fix attempt, but looks like it wasn't completely reverted, I will revert this also